scripts/package-msi.sh

```diff
@@ -17,6 +17,18 @@ cp target/artifacts/vector-"${ARCHIVE_VERSION}"-x86_64-pc-windows-msvc.zip targe
 pushd target/msi-x64
 # shellcheck disable=SC2016
 powershell '$progressPreference = "silentlyContinue"; Expand-Archive vector-'"$ARCHIVE_VERSION"'-x86_64-pc-windows-msvc.zip'
-./build.sh "${ARCHIVE_VERSION}"
+
+# building the MSI package requires the version to be purely numerical (eg 0.0.0),
+# which is not the case if with custom build workflow.
+CHANNEL="${CHANNEL:-"$(cargo vdev release channel)"}"
+
+if [[ "$CHANNEL" == "custom" ]]; then
+    # "0.29.0.custom.a28ecdc" -> "0.29.0"
```

Comment:
Why do we need to strip this?

Reply:
This change was because of this failure, which I sort of reference on line 22 (chose that line because CHANNEL is only needed because of this logic).

```
vector.wxs
C:\a\vector\vector\target\msi-x64\vector.wxs(6) : error CNDL0108 : The Product/@Version attribute's value, '0.29.0.custom.1bce710', is not a valid version.  Legal version values should look like 'x.x.x.x' where x is an integer from 0 to 65534.
```

https://github.com/vectordotdev/vector/actions/runs/4367571927/jobs/7639109583

Reply:
Ah, gotcha. Maybe we could leave that as a comment?